recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-3

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+#
+# optionally provide flashing whole disk image in INITRAMFS context
+#
+
+# Flash Mode
+#     3    flash disk image from network to active block device
+#
+OMNECT_FLASH_MODE=""
+
+OMNECT_FLASH_MODE_VAR="flash-mode"
+OMNECT_FLASH_MODE_FLAG_FILE="/etc/enforce_flash_mode_3"
+OMNECT_FLASH_MODE_FIFO="/home/omnect/wic.xz"
+OMNECT_FLASH_MODE_URL_VAR="flash-mode-url"
+
+
+# called by generic INITRAMFS
+flash_mode_3_enabled() {
+    debug "flash mode 3: ${OMNECT_FLASH_MODE}"
+
+    # flag file (part of "omnect-os-initramfs-test")
+    if [ -f ${OMNECT_FLASH_MODE_FLAG_FILE} ]; then
+        msg "enable flash mode: by ${OMNECT_FLASH_MODE_FLAG_FILE}"
+        OMNECT_FLASH_MODE=3
+        return 0  # enable flash mode; call main entry point
+    fi
+
+    # flash mode triggered at runtime
+    OMNECT_FLASH_MODE=$(get_bootloader_env_var ${OMNECT_FLASH_MODE_VAR})
+    if [ "${OMNECT_FLASH_MODE}" == "3" ]; then
+        return 0  # enable flash mode; call main entry point
+    fi
+
+    return 1  # disable flash mode; normal boot; main entry point _NOT_ called
+}
+
+run_bmap_tool() {
+    local url="${1}"
+    local umount_dev=""
+    local blk_dev=""
+    local bmap_ret=""
+
+    # detect block device
+    blk_dev=$(readlink -f /dev/omnect/boot)
+    blk_dev=${blk_dev%p*}
+    if [ ! -b "${blk_dev}" ]; then msg_fatal "${blk_dev} does not exist or not a block device"; return 1; fi
+
+    # umount, if necessary
+    mkdir -p /tmp
+    cp /proc/mounts /tmp/all_mounts  # make snapshot
+    while read x; do
+        umount_dev=$(echo $x | awk '{print $1}')
+        echo ${umount_dev} | grep -q "$(basename ${blk_dev})"
+        if [ $? -eq 0 ]; then
```

Comment:
jfi: could be condensed into 'if [[ ${umount_dev} =~ /${blkdev##*/} ]]; then ,,,'

Reply:
done